src/test_typing_extensions.py

```diff
@@ -3986,5 +3986,23 @@ def __buffer__(self, flags: int) -> memoryview:
         self.assertIsSubclass(MySubclassedBuffer, Buffer)
 
 
+class TestGenericAliasLike(BaseTestCase):
+    def test(self):
+
+        T = TypeVar('T')
+
+        class GenericType(Generic[T]):
+            def __class_getitem__(cls, args: Union[Tuple[Type[Any], ...], Type[Any]]) -> 'GenericAliasLike':
+                return GenericAliasLike()
+
+        class GenericAliasLike:
+            __args__ = ()
+            __parameters__ = (T,)
+            __origin__ = GenericType
```

Comment:
I'm hardcoding here for simplicity, but this would be some sort of dynamic generation of `__parameters__` and such.